oak/server/rust/oak_runtime/src/node/grpc/server/mod.rs

```diff
@@ -388,59 +399,65 @@ impl UnaryService<Vec<u8>> for GrpcRequestHandler {
                 request.get_ref().len(),
                 oak_label
             );
-
-            // Create a gRPC request.
-            // TODO(#953): Add streaming support.
+            // Create an encapsulated gRPC request.
+            // TODO(#97): Add client-streaming support.
             let grpc_request = GrpcRequest {
                 method_name: handler.method_name.to_string(),
                 req_msg: request.into_inner(),
                 last: true,
             };
 
-            let response = handler
-                // Handle a gRPC request and send it into the Runtime.
-                .handle_grpc_request(grpc_request, &oak_label)
-                // Read a gRPC response from the Runtime.
-                .and_then(|response_reader| handler.handle_grpc_response(response_reader))
-                // Convert an error to a gRPC error status without sending clients descriptions for
-                // internal errors.
-                // Errors are logged inside inside [`GrpcRequestHandler::handle_grpc_request`] and
-                // [`GrpcRequestHandler::handle_grpc_response`].
+            // Inject the encapsulated gRPC request into the Oak Application.
+            debug!("inject encapsulated request into Oak Node");
+            let response_iter = handler
+                .inject_grpc_request(grpc_request, &oak_label, metrics_data.clone())
                 .map_err(|()| tonic::Status::new(tonic::Code::Internal, ""))?;
 
-            // Send a gRPC response back to the client.
-            debug!("Sending gRPC response: {:?}", response);
-            timer.observe_duration();
-            match response.status {
-                None => Ok(tonic::Response::new(response.rsp_msg)),
-                Some(status) if status.code == oak_abi::proto::google::rpc::Code::Ok as i32 => {
-                    Ok(tonic::Response::new(response.rsp_msg))
+            // First convert the `Iterator<Item = GrpcResponse>` to an
+            // `Iterator<Item = Result<Vec<u8>, tonic::Status>`.
+            let result_iter = response_iter.map(|response| {
+                debug!("Sending gRPC response: {:?}", response);
```

Comment:
Is it sending a gRPC response here, or just remapping?

Reply:
Reworded to "Returning".